tests/MongoDB.Driver.Core.Tests/Core/Authentication/SaslPrepHelperTests.cs

```diff
@@ -0,0 +1,153 @@
+﻿/* Copyright 2018–present MongoDB Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using FluentAssertions;
+using MongoDB.Bson.TestHelpers;
+using Xunit;
+using MongoDB.Bson.TestHelpers.XunitExtensions;
+namespace MongoDB.Driver.Core.Authentication
+{
+    /// <summary>
+    /// SaslPrep unit tests.
+    /// </summary>
+    public class SaslPrepHelperTests
+    {
+        private static class SaslPrepHelperReflector
```

Comment:
I usually put the reflector at the bottom of the file. FYI it normally can't be a nested class because extension methods can't be defined in a nested class. It's somewhat coincidence that you didn't need to use extension methods in this reflector.

Also, it's normally a public static class.  This is a test project, we're not so concerned about what's public or not.

Reply:
Fixed!